haystack/nodes/file_converter/parsr.py

```diff
@@ -0,0 +1,237 @@
+            return ""
+        if self.remove_page_footers and "isFooter" in element["properties"]:
+            return ""
+        if element["type"] == "table-of-contents":
+            if self.remove_table_of_contents:
+                return ""
+            else:
+                current_paragraph = "\n".join([self._get_paragraph_string(elem) for elem in element["content"]])
+                return current_paragraph
+
+        current_paragraph = self._get_paragraph_string(element)
+        return current_paragraph
+
+    def _convert_table_element(self, element: Dict[str, Any], all_pages: List[Dict], page_idx: int, elem_idx: int,
+                               meta: Optional[Dict[str, str]] = None) -> Dict[str, Any]:
+        row_idx_start = 0
+        caption = ""
+        table_list = [[""] * len(element["content"][0]["content"])
+                      for _ in range(len(element["content"]))]
+
+        for row_idx, row in enumerate(element["content"]):
+            for col_idx, cell in enumerate(row["content"]):
+                # Check if first row is a merged cell spanning whole table
+                # -> exclude this row and use as caption
+                if (row_idx == col_idx == 0) and (cell["colspan"] == len(table_list[0])):
+                    cell_paragraphs = [self._get_paragraph_string(par) for par in cell["content"]]
+                    cell_content = "\n\n".join(cell_paragraphs)
+                    caption = cell_content
+                    row_idx_start = 1
+                    table_list.pop(0)
+                    break
+
+                if cell["type"] == "table-cell":
+                    cell_paragraphs = [self._get_paragraph_string(par) for par in cell["content"]]
+                    cell_content = "\n\n".join(cell_paragraphs)
+                    for c in range(cell["colspan"]):
+                        for r in range(cell["rowspan"]):
+                            table_list[row_idx + r - row_idx_start][col_idx + c] = cell_content
+
+        # Get preceding and following elements of table
+        preceding_elements = []
+        following_elements = []
+        for cur_page_idx, cur_page in enumerate(all_pages[:page_idx]):
+            for cur_elem_index, elem in enumerate(cur_page["elements"]):
+                if (elem["type"] in ["paragraph", "heading"]) \
+                        and (self.remove_page_headers and "isHeader" not in elem["properties"]) \
+                        and (self.remove_page_footers and "isFooter" not in elem["properties"]):
+                    if cur_page_idx < page_idx:
+                        preceding_elements.append(elem)
+                    elif cur_page_idx == page_idx:
+                        if cur_elem_index < elem_idx:
+                            preceding_elements.append(elem)
+                        elif cur_elem_index > elem_idx:
+                            following_elements.append(elem)
+                    elif cur_page_idx > page_idx:
```

Comment:
We only enumerate `all_pages[:page_idx]` so I am wondering when `elif cur_page_idx == page_idx:` and `elif cur_page_idx > page_idx:`become true? Am I missing something here?

Reply:
Good catch, this was remaining from when I had separate for-loops for the following and the preceding context. Fixed.